build.py

```diff
@@ -359,6 +362,99 @@ def RunMakeZipArchive(context):
+    # in FindMayaQt.cmake.
+    dirsToSearch = [context.devkitLocation]
+    if 'MAYA_DEVKIT_LOCATION' in os.environ:
+        dirsToSearch.append(os.path.expandvars('$MAYA_DEVKIT_LOCATION'))
+    dirsToSearch.append(context.mayaLocation)
+    if 'MAYA_LOCATION' in os.environ:
+        dirsToSearch.append(os.path.expandvars('$MAYA_LOCATION'))
+
+    # Check if the Qt zip file has been extracted (we need the Qt headers).
+    for dirToSearch in dirsToSearch:
+        if haveQtHeaders(dirToSearch):
+            PrintStatus('Found Maya Qt headers in: {dir}'.format(dir=dirToSearch))
+            return
+
+    # Qt5
+    # Didn't find Qt headers, so try and extract qt_5.x-include.zip (the first one we find).
+    qtIncludeArchiveName = 'qt_5*-include.zip' if Windows() else 'qt_5*-include.tar.gz'
+    for dirToSearch in dirsToSearch:
+        qtZipFiles = glob(os.path.join(dirToSearch, 'include', qtIncludeArchiveName))
+        if qtZipFiles:
+            qtZipFile = qtZipFiles[0]
+            baseDir = os.path.dirname(qtZipFile)
+            if os.access(baseDir, os.W_OK):
+                qtZipDirName = os.path.basename(qtZipFile)
+                qtZipDirName = qtZipDirName.replace('.zip', '').replace('.tar.gz', '')
+                qtZipDirFolder = os.path.join(baseDir, qtZipDirName)
+                PrintStatus("Could not find Maya Qt headers.")
+                PrintStatus("  Extracting '{zip}' to '{dir}'".format(zip=qtZipFile, dir=qtZipDirFolder))
+                if not os.path.exists(qtZipDirFolder):
+                    os.makedirs(os.path.join(baseDir, qtZipDirFolder))
+                try:
+                    # We only need certain Qt components so we only extract the headers for those.
+                    if Windows():
+                        archive = zipfile.ZipFile(qtZipFile, mode='r')
+                        files = [n for n in archive.namelist()
+                            if (n.startswith('QtCore/') or n.startswith('QtGui/') or n.startswith('QtWidgets/'))]
+                    else:
+                        archive = tarfile.open(qtZipFile, mode='r')
+                        files = [n for n in archive.getmembers()
+                            if (n.name.startswith('./QtCore/') or n.name.startswith('./QtGui/') or n.name.startswith('./QtWidgets/'))]
+                    archive.extractall(qtZipDirFolder, files)
+                    archive.close()
+                except zipfile.BadZipfile as error:
+                    PrintError(str(error))
+                except tarfile.TarError as error:
+                    PrintError(str(error))
+
+            # We found and extracted the Qt5 include zip - we are done.
+            return
+
+    # Qt6
+    # The entire Qt is in a single zip file, which we extract to 'Qt'.
+    # Then we can simply use find_package(Qt6) on it.
+    for dirToSearch in dirsToSearch:
+        # Qt archive has same name on all platforms.
+        qtArchive = os.path.join(dirToSearch, 'Qt.tar.gz')
```

Comment:
For Qt6 extract the entire archive. This one is new and contains everything needed to build with Qt.